src/collection.rs

```diff
@@ -204,10 +280,39 @@ impl<'a, C: ClientExt> Collection<'a, C> {
     /// # Note
     /// this function would make a request to arango server.
     #[maybe_async]
-    pub async fn checksum(&self) {
-        unimplemented!()
+    pub async fn checksum(&self) -> Result<CollectionChecksum, ClientError> {
+        self.checksum_with_options(false, false).await
     }
 
+    // By setting the optional query parameter withRevisions to true, then revision
+    // ids (_rev system attributes) are included in the checksumming.
+    //
+    // By providing the optional query parameter withData with a value of true, the
+    // user-defined document attributes will be included in the calculation too.
+    // Note: Including user-defined attributes will make the checksumming slower.
+    #[maybe_async]
+    pub async fn checksum_with_options(
+        &self,
+        with_revisions: bool,
+        with_data: bool,
+    ) -> Result<CollectionChecksum, ClientError> {
+        let mut query_parameters = "".to_owned();
+
+        if with_revisions == true || with_data == true {
```

Comment:
Here what we want is to add query to the requested url. So we may consider using the [set_query](https://docs.rs/url/2.1.1/url/struct.Url.html#method.set_query) method in `url::Url`.

```rust
let mut url = self.base_url.join("checksum").unwrap();
if with_revisions{
    url.set_query(Some("withRevisions=true"));
}
if with_data {
    url.set_query(Some("withData=true"));
}
// make request and serialize response
...
```

Reply:
hey ! Thank you ! I do not know why I did not use it in the first place.
It seems that set_query resets the query from the beginning, so I had to use `url.query_pairs_mut()` instead.

Reply:
Wow, I didn't know that. Thank you.

Actually `with_revisions` and `with_data` is bool, so we don't need to explicitly compare them to a bool. That's fine, never mind.